feat: add TableNodeCell component

stone-lyl opened the pull request:
Long strings/JSON should be truncated

| Before | After |
| ------ | ----- |
| ❌      | ✅    |
|![image](https://github.com/ajthinking/data-story/assets/20497176/6debf92e-dc34-4574-b56c-de9fe2f7185e)|![image](https://github.com/ajthinking/data-story/assets/20497176/b894a896-c10c-4049-90e7-7641735e221a)|

 New lines are not preserved (might be good default to keep table compact). Consider showing full/linebreaked content on cell/row click

https://github.com/ajthinking/data-story/assets/20497176/97276d9d-624a-4113-9134-f1955aee1374

ajthinking commented on packages/ui/src/components/Node/ItemCollection.test.ts:
```diff
@@ -0,0 +1,110 @@
+  it('should correctly extract headers and rows from nested JSON data', () => {
+    const { headers, rows } = new ItemCollection([nested]).toTable();
+
+    expect(headers).toMatchInlineSnapshot(`
```
Prefer toEqual per #165 I assume this is fixed once that merges 🙂

ajthinking reviewed (commented):
Super happy about this one, thank you 🙏 

1. Is the video a previous one by mistake?

2. Somethings a bit off with border styling here:
<img width="540" alt="image" src="https://github.com/ajthinking/data-story/assets/3457668/2e936dd7-b521-4d60-ba3e-4cd644ff1d49">

3. Can you take a look at the "empty state" as well? It currently just highjacks the table header/rows which is a bit weird 😅 

<img width="609" alt="image" src="https://github.com/ajthinking/data-story/assets/3457668/ff8a2abc-b3e6-41b4-acd6-19da920bfea8">


4. How to copy; case short cell, case truncated cell? Via tooltop/on click? Please think about a solution for this.

ajthinking reviewed (commented):
🟢 Preview mode:
<img width="141" alt="image" src="https://github.com/ajthinking/data-story/assets/3457668/14cfe5d8-4c0a-47ae-9e68-4b4fe9fb697c">

🟡 Tooltip mode:
<img width="273" alt="image" src="https://github.com/ajthinking/data-story/assets/3457668/2dc10891-34dd-4994-92de-7e773c91e6b7">

Can we make this pretty formatting (2 spaces)? That is: inline( truncated) just as it is now. When clicked, show a BIG linebreaked/indented JSON. I think I missed this one in requirements.

ajthinking commented:
Render boolean bug?
<img width="1654" alt="image" src="https://github.com/ajthinking/data-story/assets/3457668/0b14fc15-12dc-4445-b52d-5bb4ed169696">

stone-lyl commented:
> Super happy about this one, thank you 🙏
> 
> 1. Is the video a previous one by mistake?
> 2. Somethings a bit off with border styling here:
> 
> <img alt="image" width="540" src="https://private-user-images.githubusercontent.com/3457668/317686143-2e936dd7-b521-4d60-ba3e-4cd644ff1d49.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MTAyMDEsIm5iZiI6MTcxMTcwOTkwMSwicGF0aCI6Ii8zNDU3NjY4LzMxNzY4NjE0My0yZTkzNmRkNy1iNTIxLTRkNjAtYmEzZS00Y2Q2NDRmZjFkNDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjlUMTA1ODIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWI2MmYyMTgwYmU3MTNkNzliM2JiNzE0MjU3MTA2ZmMwN2QxYTE1OTQxOTFhNzFiNDZmYTRjMWY4ZmY5NzA0YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cQtEwC8mU6Y_fZBIa89Hfpq6LEDUl5sY3PdHR44wrwY">
> 3. Can you take a look at the "empty state" as well? It currently just highjacks the table header/rows which is a bit weird 😅
> 
> <img alt="image" width="609" src="https://private-user-images.githubusercontent.com/3457668/317687573-ff8a2abc-b3e6-41b4-acd6-19da920bfea8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3MTAyMDEsIm5iZiI6MTcxMTcwOTkwMSwicGF0aCI6Ii8zNDU3NjY4LzMxNzY4NzU3My1mZjhhMmFiYy1iM2U2LTQxYjQtYWNkNi0xOWRhOTIwYmZlYTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjlUMTA1ODIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTM5ZmU2NWU2MmQ5YjM2ZmZiY2ViNTY5ZmVhNjlkYzQxZjYyNGZkYzQ2MDA5ZDI3NDk0NjQ3OTg1MGJiMmJmMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6j3qI5T5IrIlYJVeeEi64lP8uJgHN5GKMKH2uPAZOE0">
> 4. How to copy; case short cell, case truncated cell? Via tooltop/on click? Please think about a solution for this.

1. Yeah, I've updated the video. 😊
2. I've fixed the issue with table content overflow. 
3. Yeah, I set the header to initialize as an empty state. This way, the issue won't occur. 
4. Added a copy button (like the picture below ) on the tooltip. What do you think of this idea? 🤔

![image](https://github.com/ajthinking/data-story/assets/20497176/57bf7124-3365-4c63-8225-377a9378d678)

ajthinking commented:
Changes looking good! Lets ship this 🥳⛴️
